### PR TITLE
provisioin SQL database for Warm Storage

### PR DESCRIPTION
**Pull request by hveiras**
connects to #59

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hveiras__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>You've already signed the contribution license agreement. Thanks!</span>
        <span>We will now validate the agreement and then real humans will evaluate your PR.</span>

TTYL, AZPRBOT;

File: provision/Provision-All.ps1 [hunk 1]
```diff
@@ -1,66 +1,83 @@
...
-    [String]$StreamAnalyticsJobName = "fabrikamstreamjob01",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[String]$ServiceBusNamespace,                                   
 
-	[Parameter (Mandatory = $true)]
-	[string]$SubscriptionName = "Azure Guidance",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[String]$StorageAccountName,
```
Comment (hanzzhang) [inline provision/Provision-All.ps1]: you need keep the validations:
[ValidateScript({  
11   -      If ($_ -cmatch "^[a-z0-9]*$") {                         # needs contain only lower case letters and numbers.  
12   -        $True  
13   -      }else {  
14   -        Throw "`n---Storage account name can only contain lowercase letters and numbers!---"  
15   -      }  
16   -    })]  
17   -	[Parameter (Mandatory = $true)]  
18   -    [String]$StorageAccountName = "fabrikamstorage01",

**Comment by hanzzhang** [comment]
Is there particular reason that you removed some validations? We need to keep those validations in my mind.

File: provision/Provision-All.ps1 [hunk 2]
```diff
@@ -1,66 +1,83 @@
...
 
-	[Parameter (Mandatory = $true)]
-    [String]$StreamAnalyticsJobName = "fabrikamstreamjob01",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[String]$ServiceBusNamespace,                                   
 
-	[Parameter (Mandatory = $true)]
-	[string]$SubscriptionName = "Azure Guidance",
+	[ValidateNotNullOrEmpty()]
```
Comment (hanzzhang) [inline provision/Provision-All.ps1]: we need the validation
```
[ValidateScript({  
If ($_ -cmatch "^[a-z0-9]*$") {
$True  
}else {  
Throw "`n---Storage account name can only contain lowercase letters and numbers!---"  
}  
})]  
```

File: provision/Provision-All.ps1 [hunk 3]
```diff
@@ -1,66 +1,83 @@
...
+	[string]$SqlDatabasePassword,             
 
-    [String]$ResourceGroupPrefix = "fabrikam",
+	[ValidateNotNullOrEmpty()]
+    [Parameter (Mandatory = $False)]
+	[String]$ResourceGroupPrefix = "fabrikam",
     
-    [ValidatePattern("^[A-Za-z0-9]$|^[A-Za-z0-9][\w-\.\/]*[A-Za-z0-9]$")] # needs to start with letter or number, and contain only letters, numbers, periods, hyphens, and underscores.
-    [String]$EventHubName = "eventhub01",                   
+	[ValidateNotNullOrEmpty()]
```
Comment (hanzzhang) [inline provision/Provision-All.ps1]: we need validation back

File: provision/Provision-All.ps1 [hunk 4]
```diff
@@ -1,66 +1,83 @@
...
+    [Parameter (Mandatory = $False)]
+	[String]$EventHubName = "eventhub01",               
     
-    [String]$ServiceBusRuleName = "ManagePolicy",      
+	[ValidateNotNullOrEmpty()]
+    [Parameter (Mandatory = $False)]
+	[String]$ServiceBusRuleName = "ManagePolicy",
```
Comment (hanzzhang) [inline provision/Provision-All.ps1]: this is not used anymore

File: provision/Provision-StorageAccount.ps1 [hunk 5]
```diff
@@ -1,27 +1,25 @@
...
 (
-	[Parameter (Mandatory = $true)]
-    [string]$SubscriptionName = "Azure Guidance",
+	[ValidateNotNullOrEmpty()]
+	[Parameter (Mandatory = $True)]
+	[string]$SubscriptionName,                  
 
-	[Parameter (Mandatory = $true)]
-     [string]$Location = "Central US",                   
+ 	[ValidateNotNullOrEmpty()]
```
Comment (hanzzhang) [inline provision/Provision-StorageAccount.ps1]: we need the put the validation back

**Comment by hveiras** [comment]
@hanzzhang I removed validations for all the parameters that are directly passed to Azure Cmdlets. From my point of view performing those in the scripts does not add much value because they are already performed server side. Having those validations there makes the readability of parameters difficult. I've been checking some examples of azure scripts and typically that kind of validations are not performed.

E.g: https://gallery.technet.microsoft.com/scriptcenter/Create-a-HDInsight-cluster-4edce5d7

**Comment by hveiras** [comment]
@hanzzhang @francischeung Added validations back.

**Comment by JohnPWSharp** [comment]
Shouldn't the Provision-SQLDatabase script create the resource group for the SQL Server if it doesn't exist? Currently I have to create the resource group manually before running Provision-All otherwise the New-AzureSqlServer command fails with a "resource group not found" exception.

**Comment by hveiras** [comment]
@JohnPWSharp yes, seems to be a bug in the scripts. Let me check it.

**Comment by hveiras** [comment]
@JohnPWSharp should be fixed now.

**Pull request merged**